src/node/internal/internal_timers_promises.ts

```diff
@@ -62,16 +64,23 @@ export async function setTimeout<T = void>(
   }
 
   const { promise, resolve, reject } = Promise.withResolvers<T>();
+  let onCancel: OnCancelCallback;
 
   const timer = timers.setTimeout(() => {
     resolve(value as T);
+    if (onCancel) {
+      signal?.removeEventListener('abort', onCancel);
+    }
   }, delay ?? 0);
```

Comment:
Not directly related to this PR but looks like Node Timeout use 1 when undefined

```suggestion
    }
  }, delay ?? 1);
```

Reply:
This won't work with since delay can still be undefined. I replaced it with ?? 1